builtin/common/vector.lua

```diff
@@ -143,3 +143,76 @@ function vector.sort(a, b)
 	return {x = math.min(a.x, b.x), y = math.min(a.y, b.y), z = math.min(a.z, b.z)},
 		{x = math.max(a.x, b.x), y = math.max(a.y, b.y), z = math.max(a.z, b.z)}
 end
+
+
+-- returns a new normalized 2d vector with random values
+function vector.random2d()
+	local phi = math.random() * 2 * math.pi
+	return {
+		x = math.cos(phi),
+		y = math.sin(phi),
+	}
+end
+
+-- returns a new normalized 3d vector with random values
+function vector.random3d()
+	local phi = math.random() * 2 * math.pi
+	local y = math.random() * 2 - 1
+	local r = math.sqrt(1 - y * y)
+	local x = r * math.cos(phi)
+	local z = r * math.sin(phi)
+	return {x = x, y = y, z = z}
+end
+
+-- returns a vector with a specific length
+function vector.set_length(v, l)
+	v = vector.normalize(v)
+	return vector.multiply(v, l)
+end
+
+-- limits the length of a vector
+function vector.limit(v, l)
+	if vector.length(v) > l then
+		return vector.set_length(v, l)
+	end
+	return v
+end
```

Comment:
Suboptimal, length is calculated twice. Should not be a problem, though; Lua overhead is probably higher than cost of square root calculating.